### PR TITLE
WFLY-10426 (WF14) Drop use of JpaAttachments#TRANSACTION_MANAGER + #TSR (also remove some unused parameters)

### PR DESCRIPTION
**Pull request by scottmarlow**
https://issues.jboss.org/browse/WFLY-10426
This is for WildFly 14.

### CODE REVIEW

**Review by dmlloyd — APPROVED** [review]

**Comment by kabir** [comment]
@scottmarlow the failures seem consistent, adding the fixme label.

**Comment by scottmarlow** [comment]
I'm verifying a fix locally, needed to add the WildFly Transaction Client module dependency to the weld jpa module.

**Comment by scottmarlow** [comment]
retest this please

**Comment by scottmarlow** [comment]
@kabir the failures are fixed, thanks!

File: feature-pack/src/main/resources/modules/system/layers/base/org/jboss/as/weld/jpa/main/module.xml [hunk 1]
```diff
@@ -45,8 +45,9 @@
         <module name="org.jboss.msc"/>
         <module name="javax.enterprise.api"/>
         <module name="org.wildfly.cdi-api-bridge"/>
+        <module name="org.wildfly.transaction.client"/>
         <module name="javax.persistence.api"/>
-        <module name="javax.transaction.api"/>
+        <module name="javax.transaction.api"/>
```
Comment (kabir) [inline feature-pack/src/main/resources/modules/system/layers/base/org/jboss/as/weld/jpa/main/module.xml]: @scottmarlow This space probably should not be there?

Comment (scottmarlow) [reply feature-pack/src/main/resources/modules/system/layers/base/org/jboss/as/weld/jpa/main/module.xml]: thanks, I missed that, removed the unneeded trailing space.

**Comment by scottmarlow** [comment]
two org.jboss.as.test.clustering.cluster.singleton failures on linux - elytron, doesn't seem related to this pr.

**Pull request merged**